source/renderer/app/components/widgets/forms/PinCode.js

```diff
@@ -32,25 +32,31 @@ export default class PinCode extends Component<Props> {
   focusKey = 0;
   add = false;
 
-  onChange = (inputValue: ?number, key: number) => {
+  onChange = (inputValue: ?number | ?string, key: number) => {
     const { value, onChange } = this.props;
+
     const inputNewValue =
-      inputValue !== null && inputValue !== undefined
+      inputValue !== null && inputValue !== undefined && !Number.isNaN(inputValue)
         ? inputValue.toString()
         : '';
-
     if (
       !Object.prototype.hasOwnProperty.call(value, key) ||
       value[key] === '' ||
-      inputNewValue === ''
```

Comment:
@aleksandardjordjeviciohk please, isolate these parts into a one or more functions and add comments, to make it easier to understand it